# 수연님 파일 사본
- 이가영 복사

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# OKT 사용자 사전

In [2]:
import os
#마운트
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [3]:
pwd

'/content'

In [4]:
#현재 경로. 보통 ./content일거에요!
os.getcwd()
#os.chdir('..')
os.chdir("./drive/MyDrive/csv모음")
#os.getcwd()

In [5]:
#konlpy 다운 및 호출
#1. Konlpy에서 Okt
!pip install konlpy
#2. konlpy 실행을 위한 자바어쩌구 다운
!apt-get install openjdk-8-jdk-headless -qq > /dev/null
#3. 필요한 설정
import os
os.environ["JAVA_HOME"] = "/usr/lib/jvm/java-8-openjdk-amd64"
from konlpy.tag import Okt

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 19.4/19.4 MB 52.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 488.6/488.6 kB 36.2 MB/s eta 0:00:00


In [6]:
okt = Okt()

In [7]:
# 사용자 사전 찾아가기 및 경로 설정 _ 여기서 오류가 날 시 코랩에서 폴더 열어서 usr,local 찾아가보시며 폴더명이 일치하는지 확인해보세요! 보통 파이썬 버전이 다릅니다.
os.chdir('/usr/local/lib/python3.10/dist-packages/konlpy/java')
os.getcwd()

'/usr/local/lib/python3.10/dist-packages/konlpy/java'

In [8]:
#jar파일을 풀어줄 폴더이름을 정합니다. 저는 temp_fix로 했어요!
os.makedirs('./temp_fix')
os.chdir('./temp_fix')
os.getcwd()

'/usr/local/lib/python3.10/dist-packages/konlpy/java/temp_fix'

In [9]:
#inflated어쩌고 쭉뜨고 맨끝에 에러 없으면 잘된겁니다!
!jar xvf ../open-korean-text-2.1.0.jar

  created: META-INF/
 inflated: META-INF/MANIFEST.MF
  created: org/
  created: org/openkoreantext/
  created: org/openkoreantext/processor/
  created: org/openkoreantext/processor/normalizer/
  created: org/openkoreantext/processor/phrase_extractor/
  created: org/openkoreantext/processor/qa/
  created: org/openkoreantext/processor/stemmer/
  created: org/openkoreantext/processor/tokenizer/
  created: org/openkoreantext/processor/tools/
  created: org/openkoreantext/processor/util/
  created: org/openkoreantext/processor/util/adjective/
  created: org/openkoreantext/processor/util/adverb/
  created: org/openkoreantext/processor/util/auxiliary/
  created: org/openkoreantext/processor/util/freq/
  created: org/openkoreantext/processor/util/josa/
  created: org/openkoreantext/processor/util/noun/
  created: org/openkoreantext/processor/util/substantives/
  created: org/openkoreantext/processor/util/typos/
  created: org/openkoreantext/processor/util/verb/
 inflated: org/openkoreantext/pr

In [10]:
# 사용자 사전 열기
with open(f"/usr/local/lib/python3.10/dist-packages/konlpy/java/temp_fix/org/openkoreantext/processor/util/noun/names.txt") as f:
    data = f.read()
data

'가몽\n가온\n갓세븐\n강새이\n게임닉가\n관우\n귀여미\n규\n김유이\n김준면\n까까런\n노컷\n누너예\n니노\n다마고치\n다이무스\n대학생\n데이브\n도요토미\n동운\n동이\n두주니\n디시인사이드\n디오\n라몹\n라스\n라옵\n멍구\n메이든\n명덕\n명량\n문민정부\n미네\n방엘리\n병헌\n붓다\n비정상회담\n빼빼로\n삼풍\n샤인온미\n성식\n성열\n세라문\n세라복\n세종대왕\n손권\n손책\n쇼미더머니\n쇼챔\n순규\n스라소니\n신동아\n신쓰패밀리\n신아라\n아베\n안상홍\n안홍준\n여누\n여랑\n여포\n연합\n오꾸닭\n요섭\n웃찾사\n원식\n유병언\n유비\n유이\n윤기형\n이나단\n이명박\n이완용\n임창용\n자괴\n자니윤\n자대련\n자유\n재중이\n전교조\n정윤회\n제갈량\n조자룡\n조조\n준면\n지오디\n지존파\n진영오\n차작가\n차트\n창섭\n챠트\n첸\n코르사주\n하무열\n하용파쿠\n혁재\n현이\n현태\n혜미\n'

In [48]:
# janhak
""" data += '한국외국어대학교\n한국외대\n안내\종합정보시스템\공고\n이수\n학기\n대하여\장학재단\n장학생\n지원불가\n자녀학자금\n다자녀\n지자체\nHUFS앱\n한국외대앱\n부모\n인문사회계열\n제출서류\n장학생신청서\n추천서\n추천소견\n총장직인\n자기소개서\n전공분야\n학습계획\n졸업후진로\n자유양식\nA4\n이내\n주민등록등본\n개인정보\n기초생활수급권자\n기초생활수급자증명서\n기초생활수급자\차상위계층\n모자가정\n부자가정\n주민센터\n한부모가정\n건강보험증\n건강보험자격확인서\n과세자\n급여생활자\n원천징수\n근로소득\근무처\n비사업자\n수상내역\n어학성적\n입상경력증명서\n봉사활동내역확인서\n심신장애증명서\n장애인등록수첩\n장애인카드\n직계가족\n임대차계약서\n학생지원팀\n무이자\n농어업인\n농어업\n융자금\n정규학기\n건강보험료\n한국장학재단\n학자금대출\n지원자격\n직전학기\n학부모\n1구간\n2구간\n3구간\4구간\n5구간\n6구간\n7구간\n8구간\n9구간\n10구간\n지원대상\n다문화가족\n다자녀가구\n소득구간\n거주자\n4년제\n정규대학\n중증\n석박사\n이공계\n직전학기\n등록금납입영수증\n등록금납입\n통장사본\n기획홍보팀\n사회복지사\n학생인재개발처\n추가등록\n등록기간\n\n법인사무처\n사업관리팀\
n서울시 동대문구 이문로 107\n등기부등본\n인감 도장\n설명회\n제고\n임대환경\n관리비\n전세\n가감율\
n수수료율\n총액\n전후\n점수\n기타사항\n발주처\n제3자\n대원빌딩\n서울특별시 테헤란로 98길 11\n근린 생활 시설\
n리모델링\nBRICs(브릭스)전공\nEU(유럽연합)전공\n동북아외교통상전공\n문화콘텐츠학전공\n국가리더전공\n세계문화예술경영전공\n디지털인문한국학전공\nAI융합전공\n데이터사이언스전공\n바이오데이터사이언스전공\n상담·UX심리 융합전공\n지역경제개발협력전공\n보험계리학전공\nCulture & Technology융합대학\n바이오메디컬공학부\n글로벌자유전공학부\n기후변화융합학부\n전략기획팀\n예산조정팀\n평가감사팀\n국책사업통합추진단\nIT개발팀(서울)\nIT인프라팀(서울)\n정보통신팀(글로벌)\n디지털서비스팀(서울∙글로벌)\n지식출판콘텐츠팀\n혁신지원팀수강자\n재무회계팀\n학부생\n재입학생\n분할납부자\n분할납부\n졸업유예\n졸업\n장학팀\n미취업자\n건강보험\n특별승인\n자격요건\n학생지원팀\n동문\n장학회\n장학금명\n학비조달\n조달\n소득구간확인서\n봉사활동확인서\n개인정보수집이용동의서\n지원기간\n제출방법\n방문접수\n장학팀\n모현읍\n외대로\n수혜내역\n수혜액\n수혜\n등록휴학생\n재학증명서\n건강보험료납부확인서\n양식없음\n종료일\n국가근로\n타대학생\n백분위\n학자금\n지원구간\n신청유형\n상반기\n하반기\n운영\n위원회\n장학금\n진흥원\n울산시\nHUFS\n양캠퍼스\n서울캠퍼스\n글로벌캠퍼스\n서울글로벌캠퍼스\n서울·글로벌캠퍼스\n우리은행\n단과대학\n졸업시험\n논문\n제적\n졸업요건\n지역학사\n지역전문가과정\n수료졸업대기\nELLT학과\n영어학과\n아랍어과\n미디어커뮤니케이션학부\n영어전공\n영미문학문화학과\n태국어과\n경제학부\nEICC학과\n영어통번역학과\n베트남어과\n국제통상학과\이란어과\n경영학부\n프랑스어학부\n인도어과\n영어교육과\n독일어과\n터키아제르바이잔어과\n프랑스어교육과\n노어과\n몽골어과\n독일어교육과\n스페인어과\n중국언어문화학부\n한국어교육과\n이탈리아어과\n중국외교통상학부\n국제학부\n포르투갈어과\n그리스불가리아학과\n일본언어문화학부\n사회과학전공\n네덜란드어과\n융합일본지역학부\nLD전공\n스칸디나비아어과\n정치외교학과\nLT전공\n말레이인도네시아어과\n행정학과\n서양어대학\n아시아언어문화대학\n중국어대학\n일본어대학\n사회과학\n상경\n경영\n국제학부\n사회과학전공\n사범대학\n교직\n실용외국어\n군입대\n입영통지서\n미비\n미충족\n등록휴학\n휴학\n전역일\n증빙서류\n1학기\n2학기\n개강\n종강\n학사일정\n행사명\n조치사항\n참고사항\n공고\n부서명\n학년도\n개설학과\n학수번호\교과목명\시간\n교강사명\n강의시간\n폐강공고\n일정\n단과대\n배부처\n문의\n동유럽학대학\n국제지역대학\n어문학관\n통번역대학\n인문대학\n경상대학\n자연과학대학\n학생회관\n공과대학\n바이오메디컬공학부\n졸업증명서\n문의\n전화\문의전화\n원스톱서비스센터\n졸업예정증명서\n편입생\n신입생\n신편입생\n아래\nQR코드\n지양하다\n비고\n입학금\n기타\n코딩존\n기초과학연구소\n정보산업공학연구소\n' """

" data += '한국외국어대학교\n한국외대\n안내\\종합정보시스템\\공고\n이수\n학기\n대하여\\장학재단\n장학생\n지원불가\n자녀학자금\n다자녀\n지자체\nHUFS앱\n한국외대앱\n부모\n인문사회계열\n제출서류\n장학생신청서\n추천서\n추천소견\n총장직인\n자기소개서\n전공분야\n학습계획\n졸업후진로\n자유양식\nA4\n이내\n주민등록등본\n개인정보\n기초생활수급권자\n기초생활수급자증명서\n기초생활수급자\\차상위계층\n모자가정\n부자가정\n주민센터\n한부모가정\n건강보험증\n건강보험자격확인서\n과세자\n급여생활자\n원천징수\n근로소득\\근무처\n비사업자\n수상내역\n어학성적\n입상경력증명서\n봉사활동내역확인서\n심신장애증명서\n장애인등록수첩\n장애인카드\n직계가족\n임대차계약서\n학생지원팀\n무이자\n농어업인\n농어업\n융자금\n정규학기\n건강보험료\n한국장학재단\n학자금대출\n지원자격\n직전학기\n학부모\n1구간\n2구간\n3구간\x04구간\n5구간\n6구간\n7구간\n8구간\n9구간\n10구간\n지원대상\n다문화가족\n다자녀가구\n소득구간\n거주자\n4년제\n정규대학\n중증\n석박사\n이공계\n직전학기\n등록금납입영수증\n등록금납입\n통장사본\n기획홍보팀\n사회복지사\n학생인재개발처\n추가등록\n등록기간\n\n법인사무처\n사업관리팀n서울시 동대문구 이문로 107\n등기부등본\n인감 도장\n설명회\n제고\n임대환경\n관리비\n전세\n가감율n수수료율\n총액\n전후\n점수\n기타사항\n발주처\n제3자\n대원빌딩\n서울특별시 테헤란로 98길 11\n근린 생활 시설n리모델링\nBRICs(브릭스)전공\nEU(유럽연합)전공\n동북아외교통상전공\n문화콘텐츠학전공\n국가리더전공\n세계문화예술경영전공\n디지털인문한국학전공\nAI융합전공\n데이터사이언스전공\n바이오데이터사이언스전공\n상담·UX심리 융합전공\n지역경제개발협력전공\n보험계리학전공\nCulture & Technology융합대학\n바이오메디컬공학부\n글로벌자유전공학부\n기후변화융합학부\n전략기획팀\n예산조정팀

In [11]:
# haksa
data += "한국외국어대학교\n한국외대\n안내\종합정보시스템\공고\n이수\n학기\n대하여\장학재단\n장학생\n지원불가\n자녀학자금\n다자녀\n지자체\nHUFS앱\n한국외대앱\n부모\n인문사회계열\n제출서류\n장학생신청서\n추천서\n추천소견\n총장직인\n자기소개서\n전공분야\n학습계획\n졸업후진로\n자유양식\nA4\n이내\n주민등록등본\n개인정보\n기초생활수급권자\n기초생활수급자증명서\n기초생활수급자\차상위계층\n모자가정\n부자가정\n주민센터\n한부모가정\n건강보험증\n건강보험자격확인서\n과세자\n급여생활자\n원천징수\n근로소득\근무처\n비사업자\n수상내역\n어학성적\n입상경력증명서\n봉사활동내역확인서\n심신장애증명서\n장애인등록수첩\n장애인카드\n직계가족\n임대차계약서\n학생지원팀\n무이자\n농어업인\n농어업\n융자금\n정규학기\n건강보험료\n한국장학재단\n학자금대출\n지원자격\n직전학기\n학부모\n1구간\n2구간\n3구간\4구간\n5구간\n6구간\n7구간\n8구간\n9구간\n10구간\n지원대상\n다문화가족\n다자녀가구\n소득구간\n거주자\n4년제\n정규대학\n중증\n석박사\n이공계\n직전학기\n등록금납입영수증\n등록금납입\n통장사본\n기획홍보팀\n사회복지사\n학생인재개발처\n추가등록\n등록기간\n\n법인사무처\n사업관리팀\
n서울시 동대문구 이문로 107\n등기부등본\n인감 도장\n설명회\n제고\n임대환경\n관리비\n전세\n가감율\
n수수료율\n총액\n전후\n점수\n기타사항\n발주처\n제3자\n대원빌딩\n서울특별시 테헤란로 98길 11\n근린 생활 시설\
n리모델링\nBRICs(브릭스)전공\nEU(유럽연합)전공\n동북아외교통상전공\n문화콘텐츠학전공\n국가리더전공\n세계문화예술경영전공\n디지털인문한국학전공\nAI융합전공\n데이터사이언스전공\n바이오데이터사이언스전공\n상담·UX심리 융합전공\n지역경제개발협력전공\n보험계리학전공\nCulture & Technology융합대학\n바이오메디컬공학부\n글로벌자유전공학부\n기후변화융합학부\n전략기획팀\n예산조정팀\n평가감사팀\n국책사업통합추진단\nIT개발팀(서울)\nIT인프라팀(서울)\n정보통신팀(글로벌)\n디지털서비스팀(서울∙글로벌)\n지식출판콘텐츠팀\n혁신지원팀수강자\n재무회계팀\n학부생\n재입학생\n분할납부자\n분할납부\n졸업유예\n졸업\n장학팀\n미취업자\n건강보험\n특별승인\n자격요건\n학생지원팀\n동문\n장학회\n장학금명\n학비조달\n조달\n소득구간확인서\n봉사활동확인서\n개인정보수집이용동의서\n지원기간\n제출방법\n방문접수\n장학팀\n모현읍\n외대로\n수혜내역\n수혜액\n수혜\n등록휴학생\n재학증명서\n건강보험료납부확인서\n양식없음\n종료일\n국가근로\n타대학생\n백분위\n학자금\n지원구간\n신청유형\n상반기\n하반기\n운영\n위원회\n장학금\n진흥원\n울산시\nHUFS\n양캠퍼스\n서울캠퍼스\n글로벌캠퍼스\n서울글로벌캠퍼스\n서울·글로벌캠퍼스\n우리은행\n단과대학\n졸업시험\n논문\n제적\n졸업요건\n지역학사\n지역전문가과정\n수료졸업대기\nELLT학과\n영어학과\n아랍어과\n미디어커뮤니케이션학부\n영어전공\n영미문학문화학과\n태국어과\n경제학부\nEICC학과\n영어통번역학과\n베트남어과\n국제통상학과\이란어과\n경영학부\n프랑스어학부\n인도어과\n영어교육과\n독일어과\n터키아제르바이잔어과\n프랑스어교육과\n노어과\n몽골어과\n독일어교육과\n스페인어과\n중국언어문화학부\n한국어교육과\n이탈리아어과\n중국외교통상학부\n국제학부\n포르투갈어과\n그리스불가리아학과\n일본언어문화학부\n사회과학전공\n네덜란드어과\n융합일본지역학부\nLD전공\n스칸디나비아어과\n정치외교학과\nLT전공\n말레이인도네시아어과\n행정학과\n서양어대학\n아시아언어문화대학\n중국어대학\n일본어대학\n사회과학\n상경\n경영\n국제학부\n사회과학전공\n사범대학\n교직\n실용외국어\n군입대\n입영통지서\n미비\n미충족\n등록휴학\n휴학\n전역일\n증빙서류\n1학기\n2학기\n개강\n종강\n학사일정\n행사명\n조치사항\n참고사항\n공고\n부서명\n학년도\n개설학과\n학수번호\교과목명\시간\n교강사명\n강의시간\n폐강공고\n일정\n단과대\n배부처\n문의\n동유럽학대학\n국제지역대학\n어문학관\n통번역대학\n인문대학\n경상대학\n자연과학대학\n학생회관\n공과대학\n바이오메디컬공학부\n졸업증명서\n문의\n전화\문의전화\n원스톱서비스센터\n졸업예정증명서\n편입생\n신입생\n신편입생\n아래\nQR코드\n지양하다\n비고\n입학금\n기타\n코딩존\n기초과학연구소\n정보산업공학연구소\n"

In [12]:
# 파일 새롭게 저장
with open("/usr/local/lib/python3.10/dist-packages/konlpy/java/temp_fix/org/openkoreantext/processor/util/noun/names.txt", 'w') as f:
    f.write(data)
data

'가몽\n가온\n갓세븐\n강새이\n게임닉가\n관우\n귀여미\n규\n김유이\n김준면\n까까런\n노컷\n누너예\n니노\n다마고치\n다이무스\n대학생\n데이브\n도요토미\n동운\n동이\n두주니\n디시인사이드\n디오\n라몹\n라스\n라옵\n멍구\n메이든\n명덕\n명량\n문민정부\n미네\n방엘리\n병헌\n붓다\n비정상회담\n빼빼로\n삼풍\n샤인온미\n성식\n성열\n세라문\n세라복\n세종대왕\n손권\n손책\n쇼미더머니\n쇼챔\n순규\n스라소니\n신동아\n신쓰패밀리\n신아라\n아베\n안상홍\n안홍준\n여누\n여랑\n여포\n연합\n오꾸닭\n요섭\n웃찾사\n원식\n유병언\n유비\n유이\n윤기형\n이나단\n이명박\n이완용\n임창용\n자괴\n자니윤\n자대련\n자유\n재중이\n전교조\n정윤회\n제갈량\n조자룡\n조조\n준면\n지오디\n지존파\n진영오\n차작가\n차트\n창섭\n챠트\n첸\n코르사주\n하무열\n하용파쿠\n혁재\n현이\n현태\n혜미\n한국외국어대학교\n한국외대\n안내\\종합정보시스템\\공고\n이수\n학기\n대하여\\장학재단\n장학생\n지원불가\n자녀학자금\n다자녀\n지자체\nHUFS앱\n한국외대앱\n부모\n인문사회계열\n제출서류\n장학생신청서\n추천서\n추천소견\n총장직인\n자기소개서\n전공분야\n학습계획\n졸업후진로\n자유양식\nA4\n이내\n주민등록등본\n개인정보\n기초생활수급권자\n기초생활수급자증명서\n기초생활수급자\\차상위계층\n모자가정\n부자가정\n주민센터\n한부모가정\n건강보험증\n건강보험자격확인서\n과세자\n급여생활자\n원천징수\n근로소득\\근무처\n비사업자\n수상내역\n어학성적\n입상경력증명서\n봉사활동내역확인서\n심신장애증명서\n장애인등록수첩\n장애인카드\n직계가족\n임대차계약서\n학생지원팀\n무이자\n농어업인\n농어업\n융자금\n정규학기\n건강보험료\n한국장학재단\n학자금대출\n지원자격\n직전학기\n학부모\n1구간\n2구간\n3구간\x04구간\n5구간\n6구간\n7구간\n8구간\n9구간\n10구간\n지원대상\n다문화가족\n다자녀가구\n소득구간\n거주자\

In [13]:
#다시압축해주기. 마찬가지로 끝에 notfoundfile에러같은거 없으면 잘된겁니다!
#안되면 pwd로 현재 경로가 temp_fix가 맞는지 확인해주세요
!jar cvf ../open-korean-text-2.1.0.jar *

added manifest
ignoring entry META-INF/
adding: META-INF/maven/(in = 0) (out= 0)(stored 0%)
adding: META-INF/maven/org.openkoreantext/(in = 0) (out= 0)(stored 0%)
adding: META-INF/maven/org.openkoreantext/open-korean-text/(in = 0) (out= 0)(stored 0%)
adding: META-INF/maven/org.openkoreantext/open-korean-text/pom.properties(in = 119) (out= 110)(deflated 7%)
adding: META-INF/maven/org.openkoreantext/open-korean-text/pom.xml(in = 9127) (out= 2208)(deflated 75%)
ignoring entry META-INF/MANIFEST.MF
adding: org/(in = 0) (out= 0)(stored 0%)
adding: org/openkoreantext/(in = 0) (out= 0)(stored 0%)
adding: org/openkoreantext/processor/(in = 0) (out= 0)(stored 0%)
adding: org/openkoreantext/processor/normalizer/(in = 0) (out= 0)(stored 0%)
adding: org/openkoreantext/processor/normalizer/KoreanNormalizer$Segment$.class(in = 2577) (out= 989)(deflated 61%)
adding: org/openkoreantext/processor/normalizer/KoreanNormalizer.class(in = 2980) (out= 2181)(deflated 26%)
adding: org/openkoreantext/processor/

# 세션 다시 시작하기

# 파일 가져오기

In [1]:
import os
#os.chdir('..')
os.getcwd()

'/content'

In [2]:
import pandas as pd

In [3]:
gongji = pd.read_csv('./drive/MyDrive/csv모음/final_merged_gongji(0501).csv')
haksa = pd.read_csv('./drive/MyDrive/csv모음/merged_haksa.csv')
hiring = pd.read_csv('./drive/MyDrive/csv모음/merged_hire.csv')
janghak = pd.read_csv('./drive/MyDrive/csv모음/merged_janghak.csv')


#파일복사
# 원본 데이터 보존 위해 copy 데이터 생성
gongji_copy = gongji.copy()
haksa_copy = haksa.copy()
hiring_copy = hiring.copy()
janghak_copy = janghak.copy()

# 패키지 가져오기

In [ ]:
gongji_copy.loc[6704:6708]

index        date                               title  \
6704      4  2018-01-03               대원빌딩 철거공사 입찰 관련 질의 응답   
6705      3  2018-01-02  2018년 상반기 10주 주말(토) 외국어교육과정 교육생 모집   
6706      2  2018-01-02        2018년 상반기 FLC 외국어교육과정 교육생 모집   
6707      1  2018-01-02  2018년 상반기 20주 주간/야간 외국어교육과정 교육생 모집   

                                                content  
6704  \n대원빌딩 철거공사 입찰 관련 질의 응답대원빌딩 철거공사 입찰과 관련한 질의사항에...  
6705  \n2018년 상반기 \n10주 \n주말(토) 외국어교육과정 교육생 \n모집기초부터...  
6706  \n2018년 상반기 FLC \n외국어교육과정 교육생 모집기초부터 고급까지! 해외연...  
6707  \n2018년 상반기 \n20주 \n주간/야간 외국어교육과정 교육생 \n모집기초부터...

In [4]:
# 패키지
!pip install git+https://github.com/lovit/textrank.git

  Cloning https://github.com/lovit/textrank.git to /tmp/pip-req-build-uaen8o09
  Running command git clone --filter=blob:none --quiet https://github.com/lovit/textrank.git /tmp/pip-req-build-uaen8o09
  Resolved https://github.com/lovit/textrank.git to commit 578c7fbbb22b24561b1bd4bab9a9ecc2dc9515e1
  Preparing metadata (setup.py) ... done
  Created wheel for textrank: filename=textrank-0.1.2-py3-none-any.whl size=11142 sha256=a5e7310f7c9b521453ebcab81a4ec06e582586bb27fea5a1b2396a60782864f7
  Stored in directory: /tmp/pip-ephem-wheel-cache-s_o5d2sj/wheels/8d/d1/ef/7c6abc0d5bbb64f01bd86b0ad967911d4b7f21bb44813b3239
Successfully built textrank


In [5]:
# 깃 클론
!git clone https://github.com/lovit/textrank.git

Cloning into 'textrank'...
remote: Enumerating objects: 232, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (7/7), done.
remote: Total 232 (delta 1), reused 1 (delta 0), pack-reused 225
Receiving objects: 100% (232/232), 1.21 MiB | 3.99 MiB/s, done.
Resolving deltas: 100% (133/133), done.


In [6]:
%cd textrank

/content/textrank


In [7]:
# 이제 리포지토리의 코드를 사용할 수 있습니다.
from textrank import KeywordSummarizer, KeysentenceSummarizer

# 예시 코드
def example_tokenize(sentence):
    return sentence.split()

In [8]:
import re

# 'janghak' 첫 번째 행 'content' 추출
str2 = janghak_copy.loc[300, 'content']
str2 = str2 + "시범 추가"
print(str2)

# 줄바꿈을 기준으로 문자열 분리하여 리스트에 저장
sents = re.split(r'\n|\.', str2)
print(sents)
# 빈 문자열 제거
sents = list(filter(None, sents))
print(sents)

# 공백만 있는 문자열 제거
sents = [sent for sent in sents if sent.strip()]
print(sents)


2021년 대학생 청소년교육지원장학금 멘토모집 안내 

아래와 같이 2021학년도 대학생 청소년교육지원장학금멘토 학생을 모집하니 관심있는 외대 학생들의많은 지원 바랍니다.

- 아 래 -

1. 모집기간 : 2021. 3. 2(화) ~ 3. 21(일)(* 3월 22일이후 신청자의 경우 선발 심사에 반영되지 않을 수 있음)

2. 신청유형 : B형(멘토 발굴형)

3. 선발기준
 가. 직전학기 성적 백분위 70점 이상인 자(학자금 지원구간 무관/신.편입생 성적기준 제외)
나.봉사역량 우수자 우대(봉사기간, 지속성)
 다. 자기소개, 지원동기, 성적 및 전공 관련 등을 종합적으로 평가하여 선발
 라. 초과학기자(9학기 이상), 휴학생 참여 불가, 타 대학생 근로장학사업(국가근로장학금, 다문화⦁탈북학생 멘토링장학금)과 중복 참여 불가

4. 선발 인원 :5명내외

5. 활동가능기간 :2021. 4. 1(목) ~ 2022. 1. 31(월)
*상황에 따라 활동 시작/종료일 상이, 2학기 재모집 가능

6. 활동가능시간 : 학기 중 주당 최대 10시간, 방학 중 주당 최대 25시간
 * 활동인원에 따라 활동시간 변동 가능.

7. 활동비 : 시간 당 11,150원(*최소활동 시간이 10시간 이상일 때 지급 가능)

8.신청방법 및 제출서류

 가.신청방법 : 한국장학재단 홈페이지 신청(첨부 참조)

 나. 제출서류 : 봉사활동증명서 등(해당자, 별도양식 없음)

9. 선발결과 :3월 27일 이전 선발자 대상으로 개별연락

10. 유의사항
 가. 우리 대학은 멘토 발굴형으로 학생이 직접 기관을 발굴하여 활동해야 하오니 사전에 협의 요망(첨부 참조:20년 자료 동일 함.)
 나. 대학생 청소년교육지원장학금 또한 장학금으로 교육비납입증명서에 반영, 해당 장학수혜금액만큼 교육비공제를 받을 수 없습니다.
 

2021. 3

장학팀
시범 추가
['', '2021년 대학생 청소년교육지원장학금 멘토모집 안내 ', '', '아래와 같이 2021학년도 대학생 청소년교육지원장학금멘토 학

In [9]:
# 키워드 요약----------------------

# 문장수가 문제가 아님 -> 4개여도 요약 안됨 .벡터화에서 문제

# KeywordSummarizer 인스턴스 생성
summarizer = KeywordSummarizer(tokenize=example_tokenize, verbose=True)

# 키워드 요약 실행
keywords = summarizer.summarize(sents)

# 출력
print(keywords)

word cooccurrence counting from 54 sents was done
trained TextRank. n words = 23
[('최대', 1.4176371976939008), ('주당', 1.4176371976939004), ('중', 1.4176371976939004), (':', 1.3951311473391197), ('대학생', 1.311622330435965), ('및', 1.073545580138466), ('참여', 1.0365091497724506), ('-', 1.0), ('가능', 1.0), ('따라', 1.0), ('3', 1.0), ('~', 1.0), ('선발', 0.8908000610728656), ('수', 0.8635818923870555), ('청소년교육지원장학금', 0.8406619384437963), ('신청방법', 0.7114738696497755), ('제출서류', 0.7114738696497755), ('성적', 0.6305294310319025), ('2021', 0.28175913699712085), ('1', 0.15000000000000002), ('나', 0.15000000000000002), ('가', 0.15000000000000002), ('4', 0.15000000000000002)]


# Okt

In [10]:
from konlpy.tag import Okt

okt = Okt()

def okt_tokenize(sent):
    words = okt.pos(sent, join=True)
    words = [w for w in words if ('/Noun' in w or '/Adjective' in w or '/Verb' in w)]
    return words

In [11]:
sents

['2021년 대학생 청소년교육지원장학금 멘토모집 안내 ',
 '아래와 같이 2021학년도 대학생 청소년교육지원장학금멘토 학생을 모집하니 관심있는 외대 학생들의많은 지원 바랍니다',
 '- 아 래 -',
 '1',
 ' 모집기간 : 2021',
 ' 3',
 ' 2(화) ~ 3',
 ' 21(일)(* 3월 22일이후 신청자의 경우 선발 심사에 반영되지 않을 수 있음)',
 '2',
 ' 신청유형 : B형(멘토 발굴형)',
 '3',
 ' 선발기준',
 ' 가',
 ' 직전학기 성적 백분위 70점 이상인 자(학자금 지원구간 무관/신',
 '편입생 성적기준 제외)',
 '나',
 '봉사역량 우수자 우대(봉사기간, 지속성)',
 ' 다',
 ' 자기소개, 지원동기, 성적 및 전공 관련 등을 종합적으로 평가하여 선발',
 ' 라',
 ' 초과학기자(9학기 이상), 휴학생 참여 불가, 타 대학생 근로장학사업(국가근로장학금, 다문화⦁탈북학생 멘토링장학금)과 중복 참여 불가',
 '4',
 ' 선발 인원 :5명내외',
 '5',
 ' 활동가능기간 :2021',
 ' 4',
 ' 1(목) ~ 2022',
 ' 1',
 ' 31(월)',
 '*상황에 따라 활동 시작/종료일 상이, 2학기 재모집 가능',
 '6',
 ' 활동가능시간 : 학기 중 주당 최대 10시간, 방학 중 주당 최대 25시간',
 ' * 활동인원에 따라 활동시간 변동 가능',
 '7',
 ' 활동비 : 시간 당 11,150원(*최소활동 시간이 10시간 이상일 때 지급 가능)',
 '8',
 '신청방법 및 제출서류',
 ' 가',
 '신청방법 : 한국장학재단 홈페이지 신청(첨부 참조)',
 ' 나',
 ' 제출서류 : 봉사활동증명서 등(해당자, 별도양식 없음)',
 '9',
 ' 선발결과 :3월 27일 이전 선발자 대상으로 개별연락',
 '10',
 ' 유의사항',
 ' 가',
 ' 우리 대학은 멘토 발굴형으로 학생이 직접 기관을 발굴하여 활동해야 하오니 사전에 협의 요망(첨부 참조:20년 자료 동일 함',
 

In [12]:
# 문장 요약-----------------------


# KeywordSummarizer 인스턴스 생성
summarizer = KeysentenceSummarizer(tokenize = okt_tokenize, min_sim = 0.3)

# 핵심 3문장 요약 실행
keysents = summarizer.summarize(sents, topk=3)

# 출력
print(keysents)

[(1, 1.3522548323550878, '아래와 같이 2021학년도 대학생 청소년교육지원장학금멘토 학생을 모집하니 관심있는 외대 학생들의많은 지원 바랍니다'), (49, 1.3122505469296342, ' 대학생 청소년교육지원장학금 또한 장학금으로 교육비납입증명서에 반영, 해당 장학수혜금액만큼 교육비공제를 받을 수 없습니다'), (0, 1.2610396908875314, '2021년 대학생 청소년교육지원장학금 멘토모집 안내 ')]


In [15]:
janghak_copy

index        date                                              title  \
0         1  2020-01-02          [공통][교외] 2020-1학기 남정장학재단 장학생 모집 안내(~1/10)   
1         2  2020-01-02       [공통] [대출] 2020년 1학기 농촌출신대학생 학자금융자(무이자) 지원 안내   
2         3  2020-01-07  [공통][교외] 2020년 상반기 시각장애대학생을 위한 [효명장학사업] 안내(~1/23)   
3         7  2020-01-08              [공통] [대출] 2020-1학기 한국장학재단 학자금대출 신청 안내   
4         4  2020-01-08         [공통] [대출] 2019년 하반기 천안시 대학생 학자금 대출이자 지원 안내   
...     ...         ...                                                ...   
1255   1256  2024-03-28              [공통][교내]2024-1 권순한 장학금 장학생 모집안내(~4/5)   
1256   1257  2024-03-28  [공통][교내]2024년 Matthew D. Lee & Katherine H. Le...   
1257   1258  2024-03-28                [공통][교내]2024년 최언도 장학금 장학생 모집(~4/12)   
1258   1260  2024-03-28            [공통][교내]2024-1 윤강로 장학금 장학생 모집 안내(~4/17)   
1259   1259  2024-03-28           [공통][교내]2024-1 故장경자 장학금 장학생 모집 안내(~4/17)   

                                                content  
0     \n[공통][교외] 2020-1학기 남정장학재단 장학생 모집 안내(~1/10)아래와...  
1     \n\n2020년 1학기 농촌출신대학생 학자금융자(무이자) 지원 안내\n\n \n사...  
2     \n2020년 상반기 시각장애대학생을 위한 [효명장학사업] 안내 실로암시각장애인복지...  
3     \n2020-1학기 한국장학재단 학자금대출 신청 안내\n \n1. 신청 및 실행 일...  
4     \n2019년 하반기 천안시 대학생 학자금 대출이자 지원 안내\n\n\n✿ 신청 기...  
...                                                 ...  
1255  \n2024년 권순한 장학금 장학생 모집 안내\n\n아래와 같이2024년 권순한(스...  
1256  \n2024년Matthew D. Lee & Katherine H. Lee장학금 장학...  
1257  \n2024년 최언도 장학금 장학생 모집 안내\n\n아래와 같이2024년 최언도(독...  
1258  \n2024-1윤강로 장학금 장학생 모집 안내\n\n아래와 같이2024-1학기 윤강...  
1259  \n2024-1故장경자 장학금 장학생 모집 안내\n\n아래와 같이2024-1학기故장...  

[1260 rows x 4 columns]

In [16]:
janghak_copy['content'].dtype

dtype('O')

#### 요약함수 정의
- 아래 함수 실행 시 쭉 길게 모든 행에 summarize content 추가

In [17]:
# 요약 함수 정의 _ 쭉 길게 summarizecontent만들어주는 함수
def summarize_content(content):
    # 'content'가 비어있는지 확인
    if not content.strip():
        return []  # 빈 리스트 반환

    # 줄바꿈과 온점을 기준으로 문자열 분리하여 리스트에 저장
    sents = re.split(r'\n|\.', content)

    # 빈 문자열 제거
    sents = list(filter(None, sents))

    # 공백만 있는 문자열 제거
    sents = [sent for sent in sents if sent.strip()]

    # 한 줄만 존재하는 문장이면 빈 리스트 반환
    if len(sents) == 1:
        return []
    # 요약 수행
    keysents = summarizer.summarize(sents, topk=3)

    # 요약된 문장 리스트로 반환
    print([item[2] for item in keysents])
    return keysents

# 'content' 컬럼을 문자열로 변환 (필요한 경우)
janghak_copy['content'] = janghak_copy['content'].astype(str)

# 'summarize_content' 컬럼 생성
janghak_copy['summarize_content'] = janghak_copy['content'].apply(summarize_content)

# 결과 출력
print(janghak_copy['summarize_content'].head())

# 데이터 타입 확인
print(janghak_copy['summarize_content'].dtype)

[' 장학생 추천서(붙임파일 참조, 남정장학재단 홈페이지 다운로드 가능, 추천소견 및 총장 직인공란으로 두고본인 작성 부분만 작성 후 제출)다', '1) 재산세 납세증명서(2018, 2019년) : 지방세 세목별 과세증명서 부, 모 각 1통(납부분 없을 경우, 납부분 없음 기재하여 발급 요청)2) 건강보험료 납부확인서(2018, 2019년) : 부, 모 각 1통(한 분만 납부하실 경우, 건강보험증 사본 또는 건강보험 자격확인서 함께 제출 요망)3) 소득세관련서류가) 소득세 과세자 중 급여 생활자의 경우: 근로소득 원천징수 영수증 부,모 각 1통(근무처 발행)나) 소득세 과세자 중 자영업자 등의 경우: 소득금액증명원 부,모 각 1통(세무서 발행)다) 소득세 미과세 자인 경우: 비사업자 사실증명 부,모 각 1통(세무서 발행)6', ' 어학성적사본, 전국대회 입상경력증명서 등)- 자원봉사활동내역 확인서- 심신장애증명서, 장애인 등록수첩 사본, 장애인 카드 사본(본인, 직계가족)- 직계가족의 장기입원 및 요양 시 병원에서 발급하는 입원확인 증명원 또는 진단서- 파산선고 결정문 및 경매(등기부등본) 증명(법원 발행)- 전월세 임대차계약서 사본7']
[' ※ 1차 신청기간 중 군에 있어 융자신청을 못한 군 제대 복학자의 경우, 재학생이어도 2차 신청기간 중 신청 가능(단, 병적증명서 등 증빙서류 제출 필요)', ' - 재학생 및 신입생군(1차) 신청자 실행기간: ’20', ' ※ 농어업에 종사하지 않고 농어촌지역에 6개월(180일)이상 단순 거주하는 학부모(보호자)의 자녀(대학생)인 경우 8구간이하만 융자 지원대상임']
[' 제출서류  1) 1차 서류(실로암시각장애인복지관 홈페이지에서 양식 다운이 가능합니다) ① 2020년 상반기 효명장학금 신청서 1부 ② 자기소개서 1부  (분량 A4 2~3P 내외, 모든 항목을 빠짐없이 기록하여 제출) ③ 개인정보 수집·활용 동의서 1부    ④ 2019년 2학기 성적증명서 1부 ⑤ 주민등록등본 1부    ⑥ 장애인복지카드 사본 1부

KeyboardInterrupt: 

In [ ]:
print(janghak_copy.loc[299, 'content'])

In [18]:
# 요약 함수 정의 _ 문장 하나만 실행하기
def summarize_content(content):
    # 'content'가 비어있는지 확인
    if not content.strip():
        return []  # 빈 리스트 반환

    # 줄바꿈과 온점을 기준으로 문자열 분리하여 리스트에 저장
    sents = re.split(r'\n|\.', content)

    # 빈 문자열 제거
    sents = list(filter(None, sents))

    # 공백만 있는 문자열 제거
    sents = [sent for sent in sents if sent.strip()]

    # 한 줄만 존재하는 문장이면 빈 리스트 반환
    if len(sents) == 1:
        return []

    # 요약 수행
    keysents = summarizer.summarize(sents, topk=3)

    # 요약된 문장 리스트로 반환
    return [item[2] for item in keysents]  # keysents의 3번째 요소가 문장이라고 가정

# 특정 셀에 함수 적용
content = janghak_copy.loc[298, 'content']
summarized_content = summarize_content(content)

# 결과 출력
print(summarized_content)

['2021년 1학기 농림축산식품부 장학금 중 「청년창업농육성장학생」을 다음과 같이 선발하고자 하니 대학생 여러분의 많은 참여를 바랍니다', '□ 지원내용 : 매학기 장학금(등록금 전액 + 학업장려금 200만원) ', '○ 신청 전 반드시 선발안내문을 통해 장학생 선발요건 및 신청 방법, 제출서류 등 세부사항 확인 요망']


#### 요약 함수 출력 끗

In [21]:
content = janghak_copy.loc[300, 'content']

In [22]:
# 에러 방지를 위한 빈 문자열 제거

sents = re.split(r'\n|\.', content)
# 빈 문자열 제거
sents = list(filter(None, sents))

# 공백만 있는 문자열 제거
sents = [sent for sent in sents if sent.strip()]

# 공백만 있는 문자열 제거
sents = [sent for sent in sents if sent.strip()]

# 한 줄만 존재하는 문장이면 빈 리스트 반환
if len(sents) == 1:
  sents =[]
# 요약 수행
keysents = summarizer.summarize(sents, topk=1)
keysents

[(1,
  1.3522548323550878,
  '아래와 같이 2021학년도 대학생 청소년교육지원장학금멘토 학생을 모집하니 관심있는 외대 학생들의많은 지원 바랍니다')]

In [27]:
sents


['2021년 대학생 청소년교육지원장학금 멘토모집 안내 ',
 '아래와 같이 2021학년도 대학생 청소년교육지원장학금멘토 학생을 모집하니 관심있는 외대 학생들의많은 지원 바랍니다',
 '- 아 래 -',
 '1',
 ' 모집기간 : 2021',
 ' 3',
 ' 2(화) ~ 3',
 ' 21(일)(* 3월 22일이후 신청자의 경우 선발 심사에 반영되지 않을 수 있음)',
 '2',
 ' 신청유형 : B형(멘토 발굴형)',
 '3',
 ' 선발기준',
 ' 가',
 ' 직전학기 성적 백분위 70점 이상인 자(학자금 지원구간 무관/신',
 '편입생 성적기준 제외)',
 '나',
 '봉사역량 우수자 우대(봉사기간, 지속성)',
 ' 다',
 ' 자기소개, 지원동기, 성적 및 전공 관련 등을 종합적으로 평가하여 선발',
 ' 라',
 ' 초과학기자(9학기 이상), 휴학생 참여 불가, 타 대학생 근로장학사업(국가근로장학금, 다문화⦁탈북학생 멘토링장학금)과 중복 참여 불가',
 '4',
 ' 선발 인원 :5명내외',
 '5',
 ' 활동가능기간 :2021',
 ' 4',
 ' 1(목) ~ 2022',
 ' 1',
 ' 31(월)',
 '*상황에 따라 활동 시작/종료일 상이, 2학기 재모집 가능',
 '6',
 ' 활동가능시간 : 학기 중 주당 최대 10시간, 방학 중 주당 최대 25시간',
 ' * 활동인원에 따라 활동시간 변동 가능',
 '7',
 ' 활동비 : 시간 당 11,150원(*최소활동 시간이 10시간 이상일 때 지급 가능)',
 '8',
 '신청방법 및 제출서류',
 ' 가',
 '신청방법 : 한국장학재단 홈페이지 신청(첨부 참조)',
 ' 나',
 ' 제출서류 : 봉사활동증명서 등(해당자, 별도양식 없음)',
 '9',
 ' 선발결과 :3월 27일 이전 선발자 대상으로 개별연락',
 '10',
 ' 유의사항',
 ' 가',
 ' 우리 대학은 멘토 발굴형으로 학생이 직접 기관을 발굴하여 활동해야 하오니 사전에 협의 요망(첨부 참조:20년 자료 동일 함',
 

In [29]:

# 요약 함수 정의
def summarize_content(index, content):
    # 'content'가 비어있는지 확인
    if not content:
        print(index, "Content is empty or only whitespace.")
        return []  # 빈 리스트 반환

    # 리스트의 항목들을 하나의 문자열로 결합
    if isinstance(content, list):
        content = ' '.join(content)

    # 'content'가 비어있는지 다시 확인 (공백제거 후)
    if not content.strip():
        print(index, "Content is empty or only whitespace after joining.")
        return []  # 빈 리스트 반환

    # 줄바꿈과 온점을 기준으로 문자열 분리하여 리스트에 저장
    sents = re.split(r'\n|\.', content)

    # 빈 문자열 제거
    sents = list(filter(None, sents))

    # 공백만 있는 문자열 제거
    sents = [sent for sent in sents if sent.strip()]

    # 문장이 없으면 빈 리스트 반환
    if not sents:
        print(index, "No valid sentences after splitting.")
        return []

    # 한 줄만 존재하는 문장이면 빈 리스트 반환 _ 이거 하니 오류 안나고 다돌아가는 것 같기도
    if len(sents) == 1:
        print(index, "Only one valid sentence.")
        return []

    return sents

# 특정 셀에 함수 적용
index = 299
content = janghak_copy.loc[index, 'content']
summarized_content = summarize_content(index, content)

# 결과 출력
print(summarized_content)


['붙임의 서류를 갖추어 21년도 제29기세아해암 학술장학재단 장학생 모집안내를 하외 외대 학생들의 많은 지원 바랍니다', '붙 임 모집요강 1부', ' 끝']


In [ ]:
keywordsents = ['붙임의 서류를 갖추어 21년도 제29기세아해암 학술장학재단 장학생 모집안내를 하외 외대 학생들의 많은 지원 바랍니다', '붙 임 모집요강 1부', ' 끝']

# 각 요소가 문자열인지 확인하고 문자열로 변환
keywordsents = [str(sent) for sent in keywordsents]

# 변환된 데이터를 summarizer.summarize 함수에 전달
keysents = summarizer.summarize(keywordsents, topk=1)


# 에러나는 행 정리

## 장학


#### 오류 리스트
- 7 /1259

1. janghak_copy.loc[299]
  - [공통][교외][교외]21학년도 제29기 세아해암 장학생 모집안내, Summarizer error: object dtype is not supported by sparse matrices

2. janghak_copy.loc[531]
  - [공통][국가]22-1 국가장학금 신청마감 D-7(12/30)까지, Summarizer error: object dtype is not supported by sparse matrices

3. janghak_copy.loc[537]
  - [공통][국가]22-1 국가장학금 신청마감 D-3(12/30)까지, Summarizer error: object dtype is not supported by sparse matrices

4. janghak_copy.loc[1065]
  - [공통][교외] 소나무 장학금 2023년 장학생 선발안내(7/17~7/25), Summarizer error: object dtype is not supported by sparse matrices

5. janghak_copy.loc[1083]
  - [공통][교외] 2023년 정읍시민장학재단 장학생 선발안내(7/17~7/26), Summarizer error: object dtype is not supported by sparse matrices

6. janghak_copy.loc[1085]
  - [공통][교외] 서울장학재단 2023년 서울희망 대학 진로 장학금 장학생 선발안내(7/24~7/31), Summarizer error: object dtype is not supported by sparse matrices

7. janghak_copy.loc[1128]
  - [공통][교외] 춘천시민장학재단 2023년도 하반기 장학생 선발공고, Summarizer error: object dtype is not supported by sparse matrices

### 코드

In [ ]:
ex = janghak_copy.loc[1128,'content']
print(ex)


붙임의 서류를 갖추어 재단법인 춘천시민장학재단의 2023년도 하반기 장학생 선발안내를 하니 한국외대 학생들의많은 지원 바랍니다.붙 임 선발 공고문 1부. 끝.



In [24]:
# 에러 인덱스까지 출력해보기


# 요약 함수 정의
def summarize_content(title, content):
    # 'content'가 비어있는지 확인
    if not content:
        return []  # 빈 리스트 반환

    # 리스트의 항목들을 하나의 문자열로 결합
    if isinstance(content, list):
        content = ' '.join(content)

    # 'content'가 비어있는지 다시 확인 (공백제거 후)
    if not content.strip():
        return []  # 빈 리스트 반환

    # 줄바꿈과 온점을 기준으로 문자열 분리하여 리스트에 저장
    sents = re.split(r'\n|\.', content)

    # 빈 문자열 제거
    sents = list(filter(None, sents))

    # 공백만 있는 문자열 제거
    sents = [sent for sent in sents if sent.strip()]

    # 문장이 없으면 빈 리스트 반환
    if not sents:
        return []

    # 한 줄만 존재하는 문장이면 빈 리스트 반환
    if len(sents) == 1:
        return []

    # 요약 수행
    try:
        keysents = summarizer.summarize(sents, topk=3)
    except ValueError as e:
        # 요약 도중 에러 발생시 빈 리스트 반환
        print(f"{title}, Summarizer error: {e}")
        return []

    # 요약된 문장 리스트로 반환
    return [item[2] for item in keysents]  # keysents의 3번째 요소가 문장이라고 가정

# 'summarize_content' 컬럼 생성
#janghak_copy['summarize_content'] = janghak_copy['content'].apply(summarize_content)
janghak_copy['summarize_content'] = janghak_copy.apply(lambda row: summarize_content(row['title'], row['content']), axis=1)


# 결과 출력
print(janghak_copy['summarize_content'])


[공통][교외][교외]21학년도 제29기 세아해암 장학생 모집안내, Summarizer error: object dtype is not supported by sparse matrices
[공통][국가]22-1 국가장학금 신청마감 D-7(12/30)까지, Summarizer error: object dtype is not supported by sparse matrices
[공통][국가]22-1 국가장학금 신청마감 D-3(12/30)까지, Summarizer error: object dtype is not supported by sparse matrices
[공통][교외] 소나무 장학금 2023년 장학생 선발안내(7/17~7/25), Summarizer error: object dtype is not supported by sparse matrices
[공통][교외] 2023년 정읍시민장학재단 장학생 선발안내(7/17~7/26), Summarizer error: object dtype is not supported by sparse matrices
[공통][교외] 서울장학재단 2023년 서울희망 대학 진로 장학금 장학생 선발안내(7/24~7/31), Summarizer error: object dtype is not supported by sparse matrices
[공통][교외] 춘천시민장학재단 2023년도 하반기 장학생 선발공고, Summarizer error: object dtype is not supported by sparse matrices
0       [ 장학생 추천서(붙임파일 참조, 남정장학재단 홈페이지 다운로드 가능, 추천소견 및...
1       [ ※ 1차 신청기간 중 군에 있어 융자신청을 못한 군 제대 복학자의 경우, 재학생...
2       [ 제출서류  1) 1차 서류(실로암시각장애인복지관 홈페이지에서 양식 다운이 가능합...
3       [(1차 등록기간에 대출 실행 시 등록금 전액으로 대출되므로 9학기 이하 수강자의 ...

In [25]:
# 인덱스 번호 검색하기
index = janghak_copy[janghak_copy['title'] == "[공통][교외] 춘천시민장학재단 2023년도 하반기 장학생 선발공고"].index
print(index)


Index([1128], dtype='int64')


In [26]:
# 인덱스 확인
janghak_copy.loc[1128]

index                                                             1131
date                                                        2023-09-06
title                            [공통][교외] 춘천시민장학재단 2023년도 하반기 장학생 선발공고
content              \n붙임의 서류를 갖추어 재단법인 춘천시민장학재단의 2023년도 하반기 장학생 선발...
summarize_content                                                   []
Name: 1128, dtype: object

In [33]:
janghak_copy

index        date                                              title  \
0         1  2020-01-02          [공통][교외] 2020-1학기 남정장학재단 장학생 모집 안내(~1/10)   
1         2  2020-01-02       [공통] [대출] 2020년 1학기 농촌출신대학생 학자금융자(무이자) 지원 안내   
2         3  2020-01-07  [공통][교외] 2020년 상반기 시각장애대학생을 위한 [효명장학사업] 안내(~1/23)   
3         7  2020-01-08              [공통] [대출] 2020-1학기 한국장학재단 학자금대출 신청 안내   
4         4  2020-01-08         [공통] [대출] 2019년 하반기 천안시 대학생 학자금 대출이자 지원 안내   
...     ...         ...                                                ...   
1255   1256  2024-03-28              [공통][교내]2024-1 권순한 장학금 장학생 모집안내(~4/5)   
1256   1257  2024-03-28  [공통][교내]2024년 Matthew D. Lee & Katherine H. Le...   
1257   1258  2024-03-28                [공통][교내]2024년 최언도 장학금 장학생 모집(~4/12)   
1258   1260  2024-03-28            [공통][교내]2024-1 윤강로 장학금 장학생 모집 안내(~4/17)   
1259   1259  2024-03-28           [공통][교내]2024-1 故장경자 장학금 장학생 모집 안내(~4/17)   

                                                content  \
0     \n[공통][교외] 2020-1학기 남정장학재단 장학생 모집 안내(~1/10)아래와...   
1     \n\n2020년 1학기 농촌출신대학생 학자금융자(무이자) 지원 안내\n\n \n사...   
2     \n2020년 상반기 시각장애대학생을 위한 [효명장학사업] 안내 실로암시각장애인복지...   
3     \n2020-1학기 한국장학재단 학자금대출 신청 안내\n \n1. 신청 및 실행 일...   
4     \n2019년 하반기 천안시 대학생 학자금 대출이자 지원 안내\n\n\n✿ 신청 기...   
...                                                 ...   
1255  \n2024년 권순한 장학금 장학생 모집 안내\n\n아래와 같이2024년 권순한(스...   
1256  \n2024년Matthew D. Lee & Katherine H. Lee장학금 장학...   
1257  \n2024년 최언도 장학금 장학생 모집 안내\n\n아래와 같이2024년 최언도(독...   
1258  \n2024-1윤강로 장학금 장학생 모집 안내\n\n아래와 같이2024-1학기 윤강...   
1259  \n2024-1故장경자 장학금 장학생 모집 안내\n\n아래와 같이2024-1학기故장...   

                                      summarize_content  
0     [ 장학생 추천서(붙임파일 참조, 남정장학재단 홈페이지 다운로드 가능, 추천소견 및...  
1     [ ※ 1차 신청기간 중 군에 있어 융자신청을 못한 군 제대 복학자의 경우, 재학생...  
2     [ 제출서류  1) 1차 서류(실로암시각장애인복지관 홈페이지에서 양식 다운이 가능합...  
3     [(1차 등록기간에 대출 실행 시 등록금 전액으로 대출되므로 9학기 이하 수강자의 ...  
4     [- 온라인(24일 18시까지 신청 가능) : 천안시 홈페이지 > 분야별 정보 > ...  
...                                                 ...  
1255  [*타 장학금 수혜내역이 있는 학생의 경우,선발 후 권순한 장학금 수혜액과 합쳐서 ...  
1256  [장학금 지원서1부(HUFS Ability다운로드), *선발시 기부자(이덕선 이사장...  
1257  [장학금 지원서1부(HUFS Ability다운로드), 아래와 같이2024년 최언도(...  
1258  [아래와 같이2024-1학기 윤강로(인도어77)장학금 장학생 모집을 하오니 해당되는...  
1259  [아래와 같이2024-1학기故장경자 장학금 장학생 모집을 하오니 해당되는 재학생들의...  

[1260 rows x 5 columns]

## 학사

- 2 / 621

- haksa_copy.loc[115]
  - [서울] 강의실 호실 안내, Summarizer error: object dtype is not supported by sparse matrices

- haksa_copy.loc[133]
  - 2021년 1학기 EU융합전공 개편/신설 안내 (카드뉴스), Summarizer error: object dtype is not supported by sparse matrices

In [13]:
haksa_copy

index        date                                     title  \
0        1  2020-01-06            [공통] 2019-2학기 이중전공 변경 배정 확정 공고   
1        2  2020-01-13                  [공통] 2020학년도 1학기 수강신청 공고   
2        3  2020-01-13     [공통] 2020학년도 전기 국제지역대학원 학석사연계과정 모집 안내   
3        4  2020-01-13       [공통] 2020학년도 전기 일반대학원 학석사연계과정 모집 안내   
4        5  2020-01-15                        [글로벌]졸업 예비사정 결과 조회   
..     ...         ...                                       ...   
617    618  2024-03-13                          [공통] 생리 유고 신청 안내   
618    619  2024-03-13  [공통] 졸업예정자(해당학기 졸업가능자)의 조기취업  유고결석 신청 안내   
619    621  2024-03-15           [글로벌]2024-1학기 폐강(2차) 공고(글로벌캠퍼스)   
620    620  2024-03-15            [서울] 2024-1학기 폐강(2차) 공고(서울캠퍼스)   
621    622  2024-03-26       [공통]2024-여름계절학기 개설희망 교과목 수요조사 실시 공고   

                                               content  
0    \n\n\n\n\n\n[공통]이중전공변경 배정 확정 공고\n \n2019학년도 제2...  
1    \n2020학년도 1학기 수강신청 공고 2020학년도 제1학기 수강신청에 대하여 아...  
2    \n2020학년도 전기 국제지역대학원 학석사연계과정 모집 안내\n \n2020학년도...  
3    \n2020학년도 전기 일반대학원 학석사연계과정 모집 안내\n\n2020학년도 전기...  
4    \n[글로벌] 졸업 예비사정 결과 조회글로벌캠퍼스 졸업(2020년 2월) 예비사정 ...  
..                                                 ...  
617  \n\n\n\n\n\n생리 유고 신청 안내\n\n\n\n\n\n  생리로 인한 유고...  
618  \n\n\n\n\n\n졸업예정자(해당학기 졸업가능자)의 조기취업유고결석 신청 안내\...  
619  \n\n\n\n\n\n2024-1학기 폐강(2차) 공고(글로벌캠퍼스)\n\n\n\n...  
620  \n\n\n\n\n\n2024-1학기 폐강(2차) 공고(서울캠퍼스)\n\n\n\n\...  
621  \n계절학기 개설희망 교과목 수요조사 실시 공고\n\n2024학년도여름계절학기 개설...  

[622 rows x 4 columns]

In [15]:
# 에러 인덱스까지 출력해보기
import re

# 요약 함수 정의
def summarize_content(title, content):
    # 'content'가 비어있는지 확인
    if not content:
        return []  # 빈 리스트 반환

    # 리스트의 항목들을 하나의 문자열로 결합
    if isinstance(content, list):
        content = ' '.join(content)

    # 'content'가 비어있는지 다시 확인 (공백제거 후)
    if not content.strip():
        return []  # 빈 리스트 반환

    # 줄바꿈과 온점을 기준으로 문자열 분리하여 리스트에 저장
    sents = re.split(r'\n|\.', content)

    # 빈 문자열 제거
    sents = list(filter(None, sents))

    # 공백만 있는 문자열 제거
    sents = [sent for sent in sents if sent.strip()]

    # 문장이 없으면 빈 리스트 반환
    if not sents:
        return []

    # 한 줄만 존재하는 문장이면 빈 리스트 반환
    if len(sents) == 1:
        return []

    # 요약 수행
    try:
        keysents = summarizer.summarize(sents, topk=3)
    except ValueError as e:
        # 요약 도중 에러 발생시 빈 리스트 반환
        print(f"{title}, Summarizer error: {e}")
        return []

    # 요약된 문장 리스트로 반환
    return [item[2] for item in keysents]  # keysents의 3번째 요소가 문장이라고 가정
    #return keysents

# 'summarize_content' 컬럼 생성
haksa_copy['summarize_content'] = haksa_copy.apply(lambda row: summarize_content(row['title'], row['content']), axis=1)


# 결과 출력
print(haksa_copy['summarize_content'])

[서울] 강의실 호실 안내, Summarizer error: object dtype is not supported by sparse matrices
2021년 1학기 EU융합전공 개편/신설 안내 (카드뉴스), Summarizer error: object dtype is not supported by sparse matrices
0      [ 미배정된 학생은 기존 이수중인 이중전공/부전공이 유지되며, 추후 이중전공 변경 ...
1      [ 수강신청 관련 FAQ질문답변제한인원 증원 여부전공각 학과에서 증원여부 결정 및 ...
2      [2020학년도 전기 국제지역대학원 학석사연계과정 모집 안내,  글로벌캠퍼스: 학사...
3      [2020학년도 전기 일반대학원 학석사연계과정 모집 안내,  글로벌캠퍼스: 학사종합...
4      [ 23(목)까지 조치 요망 1) 이수구분 확인 및 수정(계절학기 포함) : 학사종...
                             ...                        
617    [  생리로 인한 유고결석 신청 방법을 아래와 같이 안내합니다,  생리로 인한 유고...
618    [졸업예정자(해당학기 졸업가능자)의 조기취업 관련 유고결석 신청 방법을 아래와 같이...
619    [ 폐강 강좌를 수강하지 않은 학생은 이 기간에 수강신청 변경을 할 수없음, 영어통...
620    [2024학년도 제1학기 서울캠퍼스 폐강(2차) 강좌를 아래와 같이 공고하오니해당 ...
621    [계절학기는 2학기 개설 교과목 대상)으로 하며 캠퍼스·학과(부)간 수요조사 참여의...
Name: summarize_content, Length: 622, dtype: object


In [27]:
# 인덱스 번호 검색하기
index = haksa_copy[haksa_copy['title'] == "2021년 1학기 EU융합전공 개편/신설 안내 (카드뉴스)"].index
print(index)

Index([133], dtype='int64')


In [28]:
haksa_copy.loc[133]

index                                                              134
date                                                        2020-11-19
title                                 2021년 1학기 EU융합전공 개편/신설 안내 (카드뉴스)
content              \n2021년 1학기부터 EU 융합전공의 과목들이 신설/개편됩니다.더 자세한 내용은...
summarize_content                                                   []
Name: 133, dtype: object

In [16]:
haksa_copy

index        date                                     title  \
0        1  2020-01-06            [공통] 2019-2학기 이중전공 변경 배정 확정 공고   
1        2  2020-01-13                  [공통] 2020학년도 1학기 수강신청 공고   
2        3  2020-01-13     [공통] 2020학년도 전기 국제지역대학원 학석사연계과정 모집 안내   
3        4  2020-01-13       [공통] 2020학년도 전기 일반대학원 학석사연계과정 모집 안내   
4        5  2020-01-15                        [글로벌]졸업 예비사정 결과 조회   
..     ...         ...                                       ...   
617    618  2024-03-13                          [공통] 생리 유고 신청 안내   
618    619  2024-03-13  [공통] 졸업예정자(해당학기 졸업가능자)의 조기취업  유고결석 신청 안내   
619    621  2024-03-15           [글로벌]2024-1학기 폐강(2차) 공고(글로벌캠퍼스)   
620    620  2024-03-15            [서울] 2024-1학기 폐강(2차) 공고(서울캠퍼스)   
621    622  2024-03-26       [공통]2024-여름계절학기 개설희망 교과목 수요조사 실시 공고   

                                               content  \
0    \n\n\n\n\n\n[공통]이중전공변경 배정 확정 공고\n \n2019학년도 제2...   
1    \n2020학년도 1학기 수강신청 공고 2020학년도 제1학기 수강신청에 대하여 아...   
2    \n2020학년도 전기 국제지역대학원 학석사연계과정 모집 안내\n \n2020학년도...   
3    \n2020학년도 전기 일반대학원 학석사연계과정 모집 안내\n\n2020학년도 전기...   
4    \n[글로벌] 졸업 예비사정 결과 조회글로벌캠퍼스 졸업(2020년 2월) 예비사정 ...   
..                                                 ...   
617  \n\n\n\n\n\n생리 유고 신청 안내\n\n\n\n\n\n  생리로 인한 유고...   
618  \n\n\n\n\n\n졸업예정자(해당학기 졸업가능자)의 조기취업유고결석 신청 안내\...   
619  \n\n\n\n\n\n2024-1학기 폐강(2차) 공고(글로벌캠퍼스)\n\n\n\n...   
620  \n\n\n\n\n\n2024-1학기 폐강(2차) 공고(서울캠퍼스)\n\n\n\n\...   
621  \n계절학기 개설희망 교과목 수요조사 실시 공고\n\n2024학년도여름계절학기 개설...   

                                     summarize_content  
0    [ 미배정된 학생은 기존 이수중인 이중전공/부전공이 유지되며, 추후 이중전공 변경 ...  
1    [ 수강신청 관련 FAQ질문답변제한인원 증원 여부전공각 학과에서 증원여부 결정 및 ...  
2    [2020학년도 전기 국제지역대학원 학석사연계과정 모집 안내,  글로벌캠퍼스: 학사...  
3    [2020학년도 전기 일반대학원 학석사연계과정 모집 안내,  글로벌캠퍼스: 학사종합...  
4    [ 23(목)까지 조치 요망 1) 이수구분 확인 및 수정(계절학기 포함) : 학사종...  
..                                                 ...  
617  [  생리로 인한 유고결석 신청 방법을 아래와 같이 안내합니다,  생리로 인한 유고...  
618  [졸업예정자(해당학기 졸업가능자)의 조기취업 관련 유고결석 신청 방법을 아래와 같이...  
619  [ 폐강 강좌를 수강하지 않은 학생은 이 기간에 수강신청 변경을 할 수없음, 영어통...  
620  [2024학년도 제1학기 서울캠퍼스 폐강(2차) 강좌를 아래와 같이 공고하오니해당 ...  
621  [계절학기는 2학기 개설 교과목 대상)으로 하며 캠퍼스·학과(부)간 수요조사 참여의...  

[622 rows x 5 columns]

In [23]:
haksa_copy.loc[300,'content']

'\n2021-겨울계절학기 등록기간 연장 및 환불 안내1. 수강료 납부 및고지서 출력등록기간12. 16(목) ~ 12. 17(금) *수강료 : 1학점당 79,000원※ 수강신청자에 한하여 납부 가능함 (미수강신청자는 납부 불가)수 강 료학점당 79,000원수강료 확인(고지서 출력)“계절학기 등록금 고지서 출력”(홈페이지 핫링크 또는 종합정보시스템)납부방법-우리은행 전국지점 납부(09:00~16:00)-고지서에 있는 본인명의 우리은행 가상계좌에 입금(가상계좌 이용시간:09:00~17:00)납부확인(붙임 파일)①실시간:우리은행 홈페이지 (https://www.wooribank.com)  또는 모바일 앱→공과금→등록금→납부내역조회→가상계좌 입력→학번으로 조회②익일:종합정보시스템 로그인→등록/장학정보 클릭→교육비납입현황 또는 등록내역유의사항①수강료 납부기간 이후에는 수강료 납부 불가, 수강내역 삭제됨②인터넷뱅킹(텔레뱅킹) 이용 시 계좌번호를 정확히 입력하여야 함※가상계좌 입금시받는사람 이름이학생이름이여야 함③일부 교과목만 선택하여 납부 불가2. 수강과목 취소 및 환불 가. 취소기간 : 12. 19(일)～2022. 1. 6(목)매일 09:00~17:00까지나. 취소방법 : 종합정보시스템 ⇒ 계절학기 환불다.수강취소한 과목의 수강료는 환불규정에 따라 환불함(2022. 1. 19(수) 이후 일괄 지급 예정)구 분수업개시일 전일수업개시일4일까지8일까지12일까지12일 경과 후환불액수강료 전액수강료의 5/6해당액수강료의 2/3해당액수강료의 1/2해당액반환하지 않음신 청마감일~ 12. 21(화)17:00까지~ 12. 27(월)17:00까지~ 12. 31(금) 17:00까지~ 2022. 1. 6(목) 17:00까지2022. 1. 6(목)17:00~※천원미만 절사라. 유의사항①수강취소는 상기 해당기간에만 가능, 수강변경은 할 수 없음②수강취소(환불신청)후 다시 수강신청 불가③부분환불 가능 :수강 취소(삭제) 교과목을 선택하여 환불 신청함※환불 신청(종합정보) 화면에서 환불 받을 교과

In [24]:
haksa_copy.loc[300,'summarize_content']

[' 계절학기 등록 관련 FAQ질문답변2개 교과목을 신청, 1개의 교과목 취소 희망(일부를 듣지 않고자 할 경우)수강료를 모두 납부한 후, 부분환불(일부 교과목만 선택하여 수강료 납부는 불가)등록을 안할 경우(신청한 모든 교과목을 듣지 않고자 할 경우)등록 마감 후 수강신청 내역을 일괄 삭제됨(미등록으로 인한 불이익은 없음)4',
 ' 유의사항①수강취소는 상기 해당기간에만 가능, 수강변경은 할 수 없음②수강취소(환불신청)후 다시 수강신청 불가③부분환불 가능 :수강 취소(삭제) 교과목을 선택하여 환불 신청함※환불 신청(종합정보) 화면에서 환불 받을 교과목 선택이 가능함3',
 ' 17(금) *수강료 : 1학점당 79,000원※ 수강신청자에 한하여 납부 가능함 (미수강신청자는 납부 불가)수 강 료학점당 79,000원수강료 확인(고지서 출력)“계절학기 등록금 고지서 출력”(홈페이지 핫링크 또는 종합정보시스템)납부방법-우리은행 전국지점 납부(09:00~16:00)-고지서에 있는 본인명의 우리은행 가상계좌에 입금(가상계좌 이용시간:09:00~17:00)납부확인(붙임 파일)①실시간:우리은행 홈페이지 (https://www']

## 공지

**코드 돌아가는데 25분 정도 걸림**

**현재 적용 함수에는 외국어 제거랑 코로나 제거 까먹고 안해둔 상태....**



##### 오류 공지 목록

- 94 / 6707

1. gongji_copy.loc[129]:
- 인도연구소 2024 국제학술대회 발표문 공모
- Summarizer error: object dtype is not supported by sparse matrices

2. gongji_copy.loc[208]:
- 중국연구소 학술대회, [제18회 청년학자국제학술대회] 개최 안내
- Summarizer error: object dtype is not supported by sparse matrices

3. gongji_copy.loc[415]:
- Online workshop for HUFS faculty-Technology for the language teacher
- Summarizer error: object dtype is not supported by sparse matrices

4. gongji_copy.loc[532]:
- Celebrating Women＇s Month ＇EXTERIORES＇ Brazilian Women In Diplomacy
- Summarizer error: object dtype is not supported by sparse matrices

5. gongji_copy.loc[720]:
- 제1회 바우하우스 콜로키움 안내
- Summarizer error: object dtype is not supported by sparse matrices

6. gongji_copy.loc[833]:
- [SW중심대학] 산학협력 프로젝트 참여인력 합격자 공지
- Summarizer error: object dtype is not supported by sparse matrices

7. gongji_copy.loc[881]
- **동일 공지 1992에도 있음 확인 필요 **
- [서울] 학생상담센터 유학생 심리상담 프로그램 안내 / Psychological Counseling for International students / 海外留学生心理咨询
- Summarizer error: object dtype is not supported by sparse matrices

8. gongji_copy.loc[994]
- [쌀람누리] 인턴채용 안내
- Summarizer error: object dtype is not supported by sparse matrices

9. gongji_copy.loc[1009]
- 2022学年度后期硕士招生简章
- Summarizer error: object dtype is not supported by sparse matrices

10. gongji_copy.loc[1187]
- 개교 68주년 기념식 안내
- Summarizer error: object dtype is not supported by sparse matrices

11. gongji_copy.loc[1413]
- Year-end Tax Adjustment for Income Earned During 2021
- Summarizer error: object dtype is not supported by sparse matrices

12. gongji_copy.loc[1689]
- [Call for Participation] The 3rd Korea-India 2030 Forum
- Summarizer error: object dtype is not supported by sparse matrices

13. gongji_copy.loc[2150]
- 한반도-북방 문화전략 포럼 웹 초청장
- Summarizer error: object dtype is not supported by sparse matrices

14. gongji_copy.loc[2397]
- 한국외대 전인교육프로그램 HUFS Life Academy 4기생 추가모집
- Summarizer error: object dtype is not supported by sparse matrices

15. gongji_copy.loc[2524]
- **동일공지 4655에도 있음**
- 러시아연구소, 이문일공칠과 인문강좌 공동 개최
- Summarizer error: object dtype is not supported by sparse matrices

16. gongji_copy.loc[2721]
- **동일 공지 4852에 있음**
- 독립출판 프로젝트 2기 모집 안내
- Summarizer error: object dtype is not supported by sparse matrices

17.
제10회 전국 대학(원)생 러시아어 토론대회 온라인 개최, Summarizer error: object dtype is not supported by sparse matrices

18.
제10회 전국 대학(원)생 러시아어 토론대회 개최, Summarizer error: object dtype is not supported by sparse matrices

19.
Precautionary measures by department, Summarizer error: object dtype is not supported by sparse matrices

20.
'2019年新型冠状病毒疾病”（COVID-19）相关防治措施, Summarizer error: object dtype is not supported by sparse matrices

21.
신종 코로나바이러스감염증 예방행동수칙, Summarizer error: object dtype is not supported by sparse matrices

22.
글로벌홀과 함게하는 남예종 DIPMX 미니 콘서트, Summarizer error: object dtype is not supported by sparse matrices


23.
전자상거래중점대학지원사업 참여자 모집, Summarizer error: object dtype is not supported by sparse matrices

24.
2019년 '커리어를 부탁해' 진로개발 특강 안내, Summarizer error: object dtype is not supported by sparse matrices

25.
제11차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

26.
제10차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

27.
[글로벌] 2019-2 생명사랑 캠페인 안내, Summarizer error: object dtype is not supported by sparse matrices

28.
[서울] '2019 학생상담센터 심리검사 워크숍-내 안의 진로' 안내, Summarizer error: object dtype is not supported by sparse matrices

29.
제9차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

30.
2019년 동백도서관 『한국외대와 함께하는 인문학 특강』, Summarizer error: object dtype is not supported by sparse matrices

31.
제8차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

32.
[글로벌] 미술치료를 통한 STRESS OUT! 프로그램 안내, Summarizer error: object dtype is not supported by sparse matrices

33.
제7차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

34.
EU 10월 콜로키움 개최, Summarizer error: object dtype is not supported by sparse matrices

35.
제6차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

36.
제5차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

37.
제3차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

38.
2019 상반기 Start Up Camp 개최 안내, Summarizer error: object dtype is not supported by sparse matrices

39.
'한국외대와 함꼐하는 시민 인문학 특강' 개최, Summarizer error: object dtype is not supported by sparse matrices

40.
[글로벌] 2019학년도 정독상 독서감상문 원고모집, Summarizer error: object dtype is not supported by sparse matrices

41.
제13차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

42.
제12차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

43.
한국외대와 함께하는 시민 인문학 특강, Summarizer error: object dtype is not supported by sparse matrices

44.
제11차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

45.
제9회 전국 대학생 러시아어 토론대회 개최, Summarizer error: object dtype is not supported by sparse matrices

46.
2019 모의유럽연합 참관인 모집, Summarizer error: object dtype is not supported by sparse matrices

47.
제10차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

48.
[글로벌] 2019-1 진로 결정을 위한 자기탐색 프로그램, Summarizer error: object dtype is not supported by sparse matrices

49.
김종구 북콘서트 '기타와 글쓰기', Summarizer error: object dtype is not supported by sparse matrices

50.
[서울] 2019 현직 선배와 함께하는 JOB 코칭데이, Summarizer error: object dtype is not supported by sparse matrices

51.
제8차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

52.
제7차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

53.
제6차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

54.
제9회 전국 대학생 러시아어 토론대회 개최, Summarizer error: object dtype is not supported by sparse matrices

55.
9회 전국 대학생 러시아어 토론대회 - 예선 참가 안내 -, Summarizer error: object dtype is not supported by sparse matrices

56.
2009년 한국외대-실크로드대학생연합 새로운 회원 모집, Summarizer error: object dtype is not supported by sparse matrices

57.
[글로벌] 2019학년도 수시입학생 학점인정 AP프로그램 수강 안내, Summarizer error: object dtype is not supported by sparse matrices

58.
혈액관리본부 통·번역 서포터즈 모집 안내, Summarizer error: object dtype is not supported by sparse matrices

59.
[서울] 보건실과 함께하는 기말고사 비타민C 이벤트, Summarizer error: object dtype is not supported by sparse matrices

60.
[공통] 2019년 상반기 20주 주간 외국어교육과정 교육생 모집안내, Summarizer error: object dtype is not supported by sparse matrices

61.
2018년 동계방학 한국외대 IT자격증 특강, Summarizer error: object dtype is not supported by sparse matrices

62.
'2018년 중남미 보건의료산업 및 통역의 이해' 특강 개최, Summarizer error: object dtype is not supported by sparse matrices

63.
[글로벌] 특강 안내 [노래번역: 통번역가의 새로운 미래], Summarizer error: object dtype is not supported by sparse matrices

64.
[공통] 2018-2 창업아이디어 경진대회 참여신청 안내, Summarizer error: object dtype is not supported by sparse matrices

65.
[글로벌] 기업/산업분석(SWOT/3C/40분석법) 세미나, Summarizer error: object dtype is not supported by sparse matrices

66.
[공통] 2018 HUFS StartUp CAMP 신청 안내, Summarizer error: object dtype is not supported by sparse matrices

67.
[공통] 우리은행 대학생 홍보대사 모집 안내, Summarizer error: object dtype is not supported by sparse matrices

68.
[공통] [Instructor] 2018 Online Human Rights · Gender Equality Education (Violence Prevention Education), Summarizer error: object dtype is not supported by sparse matrices

69.
자양 라이프아카데미 3기 모집, Summarizer error: object dtype is not supported by sparse matrices

70.
2018년 대학도서관 이용자 만족도 조사 참여 안내, Summarizer error: object dtype is not supported by sparse matrices

71.
[글로벌] [18-2] 학점인정 방과후 프로그램(10주) 신청 안내, Summarizer error: object dtype is not supported by sparse matrices

72.
[공통] 2018학년도 하반기 창업동아리 모집, Summarizer error: object dtype is not supported by sparse matrices

73.
[서울] 2018년 여름방학 외대 IT자격증특강 안내, Summarizer error: object dtype is not supported by sparse matrices

74.
[마이나비코리아] 일본취업설명회 교내 개최 안내, Summarizer error: object dtype is not supported by sparse matrices

75.
2018년 제 3기 <꽃 보다 대만학교> 개최 안내, Summarizer error: object dtype is not supported by sparse matrices

76.
제8회 러시아어 토론대회 개최, Summarizer error: object dtype is not supported by sparse matrices

77.
[서울] 2018년 5월 한국외대 IT자격증특강 안내, Summarizer error: object dtype is not supported by sparse matrices

78.
2018년 러시아어 토론대회 - 예선참가 안내 -, Summarizer error: object dtype is not supported by sparse matrices

79.
학군사관(ROTC)59ㆍ60기 후보생모집안내, Summarizer error: object dtype is not supported by sparse matrices

80.
[글로벌] [18-1] 학점인정 방과후 프로그램(10주) 신청 안내, Summarizer error: object dtype is not supported by sparse matrices

81.
[서울] 2018년 3월 한국외대 IT자격증특강 안내, Summarizer error: object dtype is not supported by sparse matrices

82.
Joint Seminar organized by Human Science /.Information & Archival Science at HUFS Graduate School and Institution for Information & Archival Science
Summarizer error: object dtype is not supported by sparse matrices

83.
VIETNAM ACADEMY OF SOCIAL SCIENCES
VIETNAM INSTITUTE FOR INDIAN AND SOUTHWEST ASIAN STUDIES
Summarizer error: object dtype is not supported by sparse matrices

84.
러시아연구소, 이문일공칠과 인문강좌 공동 개최, Summarizer error: object dtype is not supported by sparse matrices
독립출판 프로젝트 2기 모집 안내, Summarizer error: object dtype is not supported by sparse matrices

85.
제10회 전국 대학(원)생 러시아어 토론대회 온라인 개최, Summarizer error: object dtype is not supported by sparse matrices

86.
Precautionary measures by department, Summarizer error: object dtype is not supported by sparse matrices

87.
'2019年新型冠?病毒疾病”（COVID-19）相?防治措施, Summarizer error: object dtype is not supported by sparse matrices

88.
제8차 러시아-CIS 전문가 초청 특강, Summarizer error: object dtype is not supported by sparse matrices

89.
[글로벌] 학생상담센터 마음돌보기 프로그램 안내, Summarizer error: object dtype is not supported by sparse matrices

90.
[글로벌] 2019학년도 수시입학생 학점인정 AP프로그램 수강 안내, Summarizer error: object dtype is not supported by sparse matrices

91.
혈액관리본부 통·번역 서포터즈 모집 안내, Summarizer error: object dtype is not supported by sparse matrices

92.
명사 특강 안내 [SNS 유명 여행작가 청춘유리], Summarizer error: object dtype is not supported by sparse matrices

93.
[공통] [Instructor] 2018 Online Human Rights · Gender Equality Education (Violence Prevention Education), Summarizer error: object dtype is not supported by sparse matrices

94.
[공통] 2018 국제심포지엄에 귀하를 초대합니다(노사발전재단), Summarizer error: object dtype is not supported by sparse matrices

### 코드

In [29]:
gongji_copy

index        date                                              title  \
0      4597  2024.03.27            [글로벌] 2024-1 명상 집단상담 ＇금모닝 마음챙김＇ 프로그램 안내   
1      4596  2024.03.27  [서울] 보건실 - 심폐소생술 교육일정 안내 (2024.04.29(월) 10시 / ...   
2      4595  2024.03.26               EU연구소 인문사회연구사업단, '제5회 바우하우스 콜로키움' 개최   
3      4594  2024.03.26                       인도연구소 HK+사업단 제66회 명사초청 특강 안내   
4      4593  2024.03.25                          FBS 외대교육방송국 78기 수습국원 추가모집   
...     ...         ...                                                ...   
6703      5  2018-01-04                경영대학원(주간․야간․주말) 2018학년도 전기 신입생 모집요강   
6704      4  2018-01-03                              대원빌딩 철거공사 입찰 관련 질의 응답   
6705      3  2018-01-02                 2018년 상반기 10주 주말(토) 외국어교육과정 교육생 모집   
6706      2  2018-01-02                       2018년 상반기 FLC 외국어교육과정 교육생 모집   
6707      1  2018-01-02                 2018년 상반기 20주 주간/야간 외국어교육과정 교육생 모집   

                                                content  
0     \n[글로벌] 2024-1 명상 집단상담 ＇금모닝 마음챙김＇ 프로그램 안내\n\n학...  
1     \n심폐소생술 교육일정 안내\n\n서울캠퍼스 보건실에서는 교내 구성원의 응급처치 능...  
2     \nEU연구소 인문사회연구사업단, '제5회 바우하우스 콜로키움' 개최\n\n\nEU...  
3     \n인도연구소 HK+사업단 제66회 명사초청 특강 안내\n\n□한국외국어대학교 인도...  
4     \nFBS 외대교육방송국 78기 수습국원 추가모집\n\n한국외국어대학교FBS외대교육...  
...                                                 ...  
6703  \n한국외국어대학교 경영대학원(주간․야간․주말) 2018학년도 전기 신입생 모집요강...  
6704  \n대원빌딩 철거공사 입찰 관련 질의 응답대원빌딩 철거공사 입찰과 관련한 질의사항에...  
6705  \n2018년 상반기 \n10주 \n주말(토) 외국어교육과정 교육생 \n모집기초부터...  
6706  \n2018년 상반기 FLC \n외국어교육과정 교육생 모집기초부터 고급까지! 해외연...  
6707  \n2018년 상반기 \n20주 \n주간/야간 외국어교육과정 교육생 \n모집기초부터...  

[6708 rows x 4 columns]

In [31]:
# 에러 인덱스까지 출력해보기
import re

# 요약 함수 정의
def summarize_content(title, content):
    # 'content'가 비어있는지 확인
    if not content:
        return []  # 빈 리스트 반환

    # 리스트의 항목들을 하나의 문자열로 결합
    if isinstance(content, list):
        content = ' '.join(content)

    # 'content'가 비어있는지 다시 확인 (공백제거 후)
    if not content.strip():
        return []  # 빈 리스트 반환

    # 줄바꿈과 온점을 기준으로 문자열 분리하여 리스트에 저장
    sents = re.split(r'\n|\.', content)

    # 빈 문자열 제거
    sents = list(filter(None, sents))

    # 공백만 있는 문자열 제거
    sents = [sent for sent in sents if sent.strip()]

    # 문장이 없으면 빈 리스트 반환
    if not sents:
        return []

    # 한 줄만 존재하는 문장이면 빈 리스트 반환
    if len(sents) == 1:
        return []

    # 요약 수행
    try:
        keysents = summarizer.summarize(sents, topk=3)
    except ValueError as e:
        # 요약 도중 에러 발생시 빈 리스트 반환
        print(f"{title}, Summarizer error: {e}")
        return []

    # 요약된 문장 리스트로 반환
    return [item[2] for item in keysents]  # keysents의 3번째 요소가 문장이라고 가정
    #return keysents

# 'summarize_content' 컬럼 생성
gongji_copy['summarize_content'] = gongji_copy.apply(lambda row: summarize_content(row['title'], row['content']), axis=1)


# 결과 출력
print(gongji_copy['summarize_content'])

인도연구소 2024 국제학술대회 발표문 공모, Summarizer error: object dtype is not supported by sparse matrices
중국연구소 학술대회, [제18회 청년학자국제학술대회] 개최 안내, Summarizer error: object dtype is not supported by sparse matrices
Online workshop for HUFS faculty-Technology for the language teacher, Summarizer error: object dtype is not supported by sparse matrices
Celebrating Women＇s Month ＇EXTERIORES＇ Brazilian Women In Diplomacy, Summarizer error: object dtype is not supported by sparse matrices
제1회 바우하우스 콜로키움 안내, Summarizer error: object dtype is not supported by sparse matrices
[SW중심대학] 산학협력 프로젝트 참여인력 합격자 공지, Summarizer error: object dtype is not supported by sparse matrices
[서울] 학생상담센터 유학생 심리상담 프로그램 안내 / Psychological Counseling for International students / 海外留学生心理咨询, Summarizer error: object dtype is not supported by sparse matrices
[쌀람누리] 인턴채용 안내, Summarizer error: object dtype is not supported by sparse matrices
2022学年度后期硕士招生简章, Summarizer error: object dtype is not supported by sparse matrices
개교 68주년 기념식 안내,

In [63]:
# 인덱스 번호 검색하기
index = gongji_copy[gongji_copy['title'] == "독립출판 프로젝트 2기 모집 안내"].index
print(index)

Index([2721, 4852], dtype='int64')


In [64]:
gongji_copy.loc[2721]

index                                                   1865
date                                              2020.08.28
title                                     독립출판 프로젝트 2기 모집 안내
content              \n독립출판 프로젝트 2기 모집 안내\n\n2020. 8.이문일공칠\n
summarize_content                                         []
Name: 2721, dtype: object

## 채용

### 오류 리스트
-  15 / 1268
- 대부분 영어라 okt가 못인식한 것

1.
Additioinal Faculty Position Announcement for Spring 2024 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices

2.
Faculty Position Announcement for Spring 2024 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices

3.
Faculty Position Announcement for Fall 2023(Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices

4.
Faculty Position Announcement for Spring 2023 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices

5.
[쌀람누리] 인턴채용 안내, Summarizer error: object dtype is not supported by sparse matrices

6.
Faculty Position Announcement for Fall 2022 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices

7.
Faculty Position Announcement for Spring 2022 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices

8.
Job Opening Announcement for HK Research Professor for HK+ ‘Great Transition in India’ Project at Institute of Indian Studies, H, Summarizer error: object dtype is not supported by sparse matrices

9.
Faculty Position Announcement for Fall 2021 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices

10.
한국외국어대학교 신입 직원 채용공고, Summarizer error: object dtype is not supported by sparse matrices

11.
한국외국어대학교 입학사정관 채용공고, Summarizer error: object dtype is not supported by sparse matrices

12.
2021 Online International Summer Session (ISS) Instructor Recruitment Annoucement, Summarizer error: object dtype is not supported by sparse matrices

13.
Faculty Position Announcement for Spring 2021 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices

14.
2020 International Youth Leader Exchange Support Program Participant Recruiting Announcement, Summarizer error: object dtype is not supported by sparse matrices

15.
Faculty Position Announcement for Fall 2020 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices

### 코드

In [66]:
# 에러 인덱스까지 출력해보기
import re

# 요약 함수 정의
def summarize_content(title, content):
    # 'content'가 비어있는지 확인
    if not content:
        return []  # 빈 리스트 반환

    # 리스트의 항목들을 하나의 문자열로 결합
    if isinstance(content, list):
        content = ' '.join(content)

    # 'content'가 비어있는지 다시 확인 (공백제거 후)
    if not content.strip():
        return []  # 빈 리스트 반환

    # 줄바꿈과 온점을 기준으로 문자열 분리하여 리스트에 저장
    sents = re.split(r'\n|\.', content)

    # 빈 문자열 제거
    sents = list(filter(None, sents))

    # 공백만 있는 문자열 제거
    sents = [sent for sent in sents if sent.strip()]

    # 문장이 없으면 빈 리스트 반환
    if not sents:
        return []

    # 한 줄만 존재하는 문장이면 빈 리스트 반환
    if len(sents) == 1:
        return []

    # 요약 수행
    try:
        keysents = summarizer.summarize(sents, topk=3)
    except ValueError as e:
        # 요약 도중 에러 발생시 빈 리스트 반환
        print(f"{title}, Summarizer error: {e}")
        return []

    # 요약된 문장 리스트로 반환
    return [item[2] for item in keysents]  # keysents의 3번째 요소가 문장이라고 가정
    #return keysents

# 'summarize_content' 컬럼 생성
hiring_copy['summarize_content'] = hiring_copy.apply(lambda row: summarize_content(row['title'], row['content']), axis=1)


# 결과 출력
print(hiring_copy['summarize_content'])

Additioinal Faculty Position Announcement for Spring 2024 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices
Faculty Position Announcement for Spring 2024 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices
Faculty Position Announcement for Fall 2023(Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices
Faculty Position Announcement for Spring 2023 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices
[쌀람누리] 인턴채용 안내, Summarizer error: object dtype is not supported by sparse matrices
Faculty Position Announcement for Fall 2022 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices
Faculty Position Announcement for Spring 2022 (Foreign Faculty), Summarizer error: object dtype is not supported by sparse matrices
Job Opening Announcement for HK Research Professor for HK+ ‘Great Transition in India’ Project at Institute of Indian 

In [68]:
# 인덱스 번호 검색하기
index = hiring_copy[hiring_copy['title'] == "Additioinal Faculty Position Announcement for Spring 2024 (Foreign Faculty)"].index
print(index)

Index([137], dtype='int64')


In [69]:
hiring_copy.loc[137, 'content']

'\nAdditioinal Faculty Position Announcement for Spring 2024 (Foreign Faculty)\n\n1. Job Openings by Departments and Specializations\n1) Global Campus (Yong-in)\n\n\n\n\nCollege\n\nDepartment\n\nOpenings\n\nSpecialization\n\n\n\nCollege of Central and East European Studies\n\nDepartment of Romanian\n\n1\n\nRomanian History\n\n\n\n\n2. How to Apply\n ○ To submit the required application documents :\n⇒ Contact the department with the job opening of your academic interest for all inquiries including the due date for the application and the general questions about the application documents.\n Please refer to the contact information at the end of this announcement.\n※ Application documents should be submitted by November 17, 2023. The exact due date varies by departments.\n※ The duration of contract is one year. The department may consider an extended contract for up to five years on the basis of strict evaluation of each applicant. Note that every contract is renewable at the time of termi